lib/Service/RecordingService.php

```diff
@@ -137,6 +142,66 @@ public function store(Room $room, string $owner, array $file): void {
 		} catch (NotPermittedException $e) {
 			throw new InvalidArgumentException('owner_permission');
 		}
+
+		try {
+			$this->speechToTextManager->scheduleFileTranscription($fileNode, $owner, Application::APP_ID);
+		} catch (PreConditionNotMetException $e) {
+			// No Speech-to-text provider installed
+			$this->logger->debug('Could not generate transcript of call recording', ['exception' => $e]);
+		} catch (InvalidArgumentException $e) {
+			$this->logger->warning('Could not generate transcript of call recording', ['exception' => $e]);
+		}
+	}
+
+	public function storeTranscript(string $owner, File $recording, string $transcript): void {
+		$recordingFolder = $recording->getParent();
+		$roomToken = $recordingFolder->getName();
+
+		try {
+			$room = $this->roomManager->getRoomForUserByToken($roomToken, $owner);
+			$participant = $this->participantService->getParticipant($room, $owner);
+		} catch (ParticipantNotFoundException) {
+			$this->logger->warning('Could not determinate conversation when trying to store transcription of call recording');
+			throw new InvalidArgumentException('owner_participant');
```

Comment:
What's with the exception message here? Is this intentional?

Reply:
its for debugging (when listeners throw in the event handling it will now log this message)